### PR TITLE
fix: :bug: Fix and rework logging

### PR DESCRIPTION
**Pull request by zarik5**
This PR achieves multiple things:
* Fix debug group filter not actually filtering logs to file or stdout
* Use log target instead of message prefix to filter debug groups
* Make debug groups separate events
* Make logging to dashboard or file more legible
* Add server impl (SteamVR) group logs

### CODE REVIEW

**Comment by The-personified-devil** [comment]
I mean I didn't catch it in code review either, but how do you keep managing to not test stuff

**Review by The-personified-devil — CHANGES_REQUESTED** [review]

File: alvr/server_openvr/cpp/alvr_server/Controller.cpp [hunk 1]
```diff
@@ -17,6 +17,8 @@ vr::ETrackedDeviceClass Controller::getControllerDeviceClass() {
 Controller::Controller(uint64_t deviceID, vr::EVRSkeletalTrackingLevel skeletonLevel)
     : TrackedDevice(deviceID)
     , m_skeletonLevel(skeletonLevel) {
+    Debug("Controller::constructor");
```
[the remark this answers is not in the document]
Comment (zarik5) [reply alvr/server_openvr/cpp/alvr_server/Controller.cpp:20]: Fair

File: alvr/server_openvr/cpp/alvr_server/Controller.cpp [hunk 2]
```diff
@@ -180,6 +184,8 @@ void Controller::RegisterButton(uint64_t id) {
 }
 
 void Controller::SetButton(uint64_t id, FfiButtonValue value) {
+    Debug("Controller::RegisterButton id=%llu", id);
```
[the remark this answers is not in the document]
Comment (zarik5) [reply alvr/server_openvr/cpp/alvr_server/Controller.cpp:187]: whops

File: alvr/server_core/src/connection.rs [hunk 3]
```diff
@@ -683,6 +683,8 @@ fn connection_pipeline(
         },
     ));
 
+    debug!(target: "server_core", "Connection established with {client_hostname}");
```
[the remark this answers is not in the document]
Comment (zarik5) [reply alvr/server_core/src/connection.rs:686]: uhm, that was me testing the intended syntax that is then abstracted away with the macros I defined in alvr_common. I forgot to reconvert this to our custom macros

File: alvr/dashboard/src/dashboard/components/logs.rs [hunk 4]
```diff
@@ -38,59 +38,37 @@ impl LogsTab {
...
-                    LogSeverity::Info => {
-                        color = log_colors::INFO_LIGHT;
-                        ty = "INFO";
-                    }
-                    LogSeverity::Debug => {
-                        color = log_colors::DEBUG_LIGHT;
-                        ty = "DEBUG";
-                    }
-                };
+        let color = if let EventType::Log(entry) = &event.event_type {
```
[the remark this answers is not in the document]
Comment (zarik5) [reply alvr/dashboard/src/dashboard/components/logs.rs:41]: Should I split it as `color` and `should_print` variables?

[the remark this answers is not in the document]
Comment (The-personified-devil) [reply alvr/dashboard/src/dashboard/components/logs.rs:41]: leave it, it's now an easter egg

**Review by The-personified-devil — APPROVED** [review]

**Pull request merged**